### PR TITLE
building conda packages

### PR DESCRIPTION
**Pull request by bottler**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #407

### CODE REVIEW

File: packaging/conda/build_conda.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+# python packaging/conda/build_conda.py  --cuda 11.6 --python 3.10 --pytorch 1.12.1
+# python packaging/conda/build_conda.py  --cuda 11.3 --python 3.9 --pytorch 1.12.1  # <= the dino one
+# python packaging/conda/build_conda.py  --cuda 11.6 --python 3.10 --pytorch 1.11.0
+
+# Note this does the build outside the root of the tree.
+
+# TODO:
+# - Make a local conda package cache available inside docker
+# - use ninja
+# - do we need builds for both _GLIBCXX_USE_CXX11_ABI values?
```
Comment (fmassa) [inline packaging/conda/build_conda.py:165]: let's not bother about this for now

File: packaging/conda/build_conda.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+# python packaging/conda/build_conda.py  --cuda 11.6 --python 3.10 --pytorch 1.12.1
+# python packaging/conda/build_conda.py  --cuda 11.3 --python 3.9 --pytorch 1.12.1  # <= the dino one
+# python packaging/conda/build_conda.py  --cuda 11.6 --python 3.10 --pytorch 1.11.0
+
+# Note this does the build outside the root of the tree.
+
+# TODO:
+# - Make a local conda package cache available inside docker
+# - use ninja
```
Comment (fmassa) [inline packaging/conda/build_conda.py:164]: this is going to be important for parallelizing the builds. I'd hope it should be as simple as getting `ninja` installed through conda beforehand

**Review by fmassa — APPROVED** [review]
This is looks great, thanks Jeremy!

We should probably look at some point in pushing the artifacts into a conda channel.

cc @blefaudeux for visibility

**Comment by fmassa** [comment]
@bottler Also, minor nit, could you look into fixing the linter?

**Comment by erip** [comment]
This will close #152

**Comment by blefaudeux** [comment]
looks great, thanks a bunch @bottler, this will be super helpful

**Comment by blefaudeux** [comment]
cc @h-vetinari to see if this can be helpful on conda's side. This could for instance help with build issues for a lot of users around stable diffusion and[ a speed up with xformers](https://github.com/huggingface/diffusers/pull/532)

**Pull request merged**